### PR TITLE
Comment system using Arrays

### PR DESCRIPTION
**Pull request by AidanMolloy**
- DataServlet holding array and adding new comments after POST
- Comments page with comments display area and form for new comments
- JS for fetching comments and displaying them in comment area

TODO: Tests & Datastore

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -12,70 +12,70 @@
...
 }
 
-/**
- * Fetches a my name from the server and adds it to the DOM.
- */
-function getName() {
-  // The fetch() function returns a Promise because the request is asynchronous.
-  const responsePromise = fetch('/data');
+// Function to fetch comments from /data
+function getComments() {
```
Comment (marvel308) [inline portfolio/src/main/webapp/script.js:60]: https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -12,70 +12,70 @@
...
 
+// Function that opens the selected content.
 function openContent(evt, page) {
-  let i, content, navitem;
+  let i;
 
-  content = document.getElementsByClassName("content");
+  // Set all the pages display to none
+  const content = document.getElementsByClassName("content");
   for (i = 0; i < content.length; i++) {
```
Comment (marvel308) [inline portfolio/src/main/webapp/script.js]: Can this be done using forEach or map?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -19,14 +19,78 @@
...
+
+  // Function to add a comment to the array
+  public static String[][] addComment(int arrayLength, String arr[][], String commentToBeAdded[])
+  {
+    int i;
+
+    // create a new array of size arrayLength+1
+    String newarr[][] = new String[arrayLength + 1][3];
+
+    // insert the elements from the old array into the new array
```
Comment (marvel308) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I'm not sure why this function is needed. Can we use an ArrayList or some other data structure so we can just push the new comment?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -19,14 +19,78 @@
...
+    // Add the new comment to Array.
+    String[] newComment = {userName, comment, dtf.format(now)};
+    comments = addComment(comments.length, comments, newComment);
+
+    // Redirect user once comment added TODO: Redirect to Comments tab.
+    response.sendRedirect("/index.html#comments");
+  }
+
+  // Function to add a comment to the array
+  public static String[][] addComment(int arrayLength, String arr[][], String commentToBeAdded[])
```
Comment (marvel308) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: https://google.github.io/styleguide/javaguide.html#s7-javadoc

**Review by marvel308 — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (AidanMolloy) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Having difficulties with ArrayLists to Gson, will come back to this once datastore is in place.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (AidanMolloy) [reply portfolio/src/main/webapp/script.js:60]: Fixed

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (AidanMolloy) [reply portfolio/src/main/webapp/script.js]: Changed to using forEach on the HTMLCollection

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (AidanMolloy) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated documentation to follow style guide

**Review by marvel308 — APPROVED** [review]

**Pull request merged**